### PR TITLE
OSX Build

### PR DESCRIPTION
**Pull request by OniDaito**
Here is my first stab at a version that works on OS X. Needs a few bits still adding. Might make a good branch :) Mostly, its supporting the older GLSL and making odd changes to the json config, altering the makefile for packaging the final app and commenting out the midi for now.

### CODE REVIEW

**Comment by OniDaito** [comment]
1 - sure! I'll need to add to the readme that the config can't contain comments or trailing commas (for some reason) and that textures and configs need to be next to executable. I shall do so.
2 - Yeah, platform_osx is a good idea.
3 - As far as I can see the MIDI is only removed for OS X; I wouldn't compile for me. I'll check again but the pragmas shouldn't affect windows

**Comment by Gargaj** [comment]
You removed the MIDI config loader code for some reason. Again, I don't recommend removing or #ifdef-ing out anything, just add a blank MIDI.cpp and always return TRUE or 1.0 on everything.

File: main.cpp [hunk 1]
```diff
@@ -151,13 +150,31 @@ int main()
       if (o.get<jsonxx::Object>("gui").has<jsonxx::Boolean>("visibleWhitespace"))
         options.bVisibleWhitespace = o.get<jsonxx::Object>("gui").get<jsonxx::Boolean>("visibleWhitespace");
     }
-    if (o.has<jsonxx::Object>("midi"))
+    
+    /*if (o.has<jsonxx::Object>("midi"))
```
Comment (Gargaj) [inline main.cpp]: See here.

**Comment by OniDaito** [comment]
Yeah, I caught that last night :) Thanks. I'll open a new pull request, though I've kept the pragmas for the includes and non-compatible Apple things in; I'd rather not include dummy versions when two lines suffices.

**Comment by Gargaj** [comment]
I'd prefer if you do the dummy versions because that keeps Apple-specific things in their own file which the rest can ignore. Littering all over the main code with #ifdef-s makes no sense to me.

**Comment by OniDaito** [comment]
Ok so the MIDI is still there, just with a blank for the OSX. I've added some support for pathing with the textures and config files (so it can run as a .app) and I've made a couple of changes to the readme. Seems to work on OS X but haven't yet had chance to test windows build. I suspect it will be fine but if I get chance, I'll give it a test.

**Comment by Gargaj** [comment]
There's still a whole lot of ```#ifdef __APPLE__``` in there instead of reimplementing the ```Misc``` stuff.

**Comment by OniDaito** [comment]
Yeah some of these are un-avoidable, but I can get rid of a few. Right now though, I've discovered the SDL is having problems with MacOSX and fullscreen which is no good at all, so looking at that first.

**Comment by OniDaito** [comment]
Yes, looking at it, SDL2 is the only real way to go on the later OS X versions. I see you've started a branch on that so I reckon I'll close this and take a look at that instead. Had a lot of issues with unicode and fullscreen but they both appear to work fine now.

**Pull request closed without merging**